### PR TITLE
[MOV] event_barcode: merge into event

### PR DESCRIPTION
**Pull request by datonefaridze**
This PR moves event_barcode module from enterprise to community. It is
now integrated directly with the base 'event' module. Using it is optional
and activated directly from settings, instead of having to install a
new module.

This PR also allows organizers to use QR codes without needing a barcode.

Some improvements are made in registration flow, notably improving barcode
display and usage in checkout process and outgoing emails.

Task-3411827 (Move event barcode to community)
Task-3470321 (Improve registration flow)

### CODE REVIEW

**Review by reth-odoo — COMMENTED** [review]
As discussed will hold off reviewing while you check the rebase, cheers!

**Review by reth-odoo — COMMENTED** [review]
Hey :wave: @datonefaridze 

Here's a first pass. It seems like everything was moved properly so far :)
I have a few general comments, and a couple functional ones.

File: addons/event/data/mail_template_data.xml
Comment (Odoonan) [reply addons/event/data/mail_template_data.xml:270]: Hello, Just passing by @thiryr , it is indeed strange but it comes from the 'Example Tickets' that can be printed on an event. This is meant to represent the complete ticket, hence that logic. (not attendee -> Display data as this means we are in dummy mode) Cheers!

**Comment by datonefaridze** [comment]
@thiryr can you please check it again?

File: addons/event/views/event_report_templates.xml
Comment (thiryr) [inline addons/event/views/event_report_templates.xml:66]: Maybe move these to `event_registration_report.xml`? It seems like we define the other templates in that folder without issue

File: addons/event/models/event_registration.py
Comment (thiryr) [inline addons/event/models/event_registration.py:88]: This now needs to be an upgrade script, as this was used to initialize a barcode for existing registrations that don't have them yet, which there won't be any now.

**Review by reth-odoo — COMMENTED** [review]
:wave: Seems mostly good for the MOV

Just a few more comments on that, reiterating on hiding the barcode for emails and we need an upgrade script :D

File: addons/event/models/res_config_settings.py
Comment (thiryr) [reply addons/event/models/res_config_settings.py:27]: -> As per our discussion, the option actually means we want to use barcodes in addition to QR codes. Not that we want to enable QR codes at all.

So this should remain to decide whether we want both qr codes and barcodes or just barcodes.

File: addons/event/views/res_config_settings_views.xml
Comment (thiryr) [inline addons/event/views/res_config_settings_views.xml:49]: This needs to reflect what we actually want to do. Not enable/disable scanning entirely, but enable/disable **bar** codes, as separate from QR codes.

Comment (datonefaridze) [reply addons/event/views/res_config_settings_views.xml:49]: @thiryr do you want me to change message?

File: addons/event/views/event_event_views.xml
Comment (datonefaridze) [reply addons/event/views/event_event_views.xml:338]: @thiryr i can't do this because things that it is reffering to are loaded second (if i move) therefore it raises an error because `event.event_barcode_action_main_view` isn't loaded yet

File: addons/event/views/res_config_settings_views.xml
Comment (thiryr) [reply addons/event/views/res_config_settings_views.xml:49]: Yes, since it doesn't actually impact scanning

File: addons/event/views/event_event_views.xml
Comment (thiryr) [reply addons/event/views/event_event_views.xml:338]: If there's no dependency the other way around, you can change the order of the files in the manifest and it will fix your ordering issue. Otherwise fine I guess :shrug:

Comment (datonefaridze) [reply addons/event/views/event_event_views.xml:338]: @thiryr i re-ordered and it seems that there is another issue, so i don't think it's a big deal to leave it there

**Comment by datonefaridze** [comment]
@thiryr i updated the code

File: addons/website_event/views/event_templates_page_registration.xml
Comment (thiryr) [inline addons/website_event/views/event_templates_page_registration.xml:87]: Same question as below

File: addons/event/report/event_registration_report.xml
Comment (thiryr) [inline addons/event/report/event_registration_report.xml:80]: This seems to be missing from the manifest.

File: addons/event/__manifest__.py
Comment (thiryr) [inline addons/event/__manifest__.py:39]: Useless change, even if correct it's best to avoid adding steps when blaming stuff imo

**Review by reth-odoo — COMMENTED** [review]
@datonefaridze  Looking pretty good to me, I think once the unresolved comments are addressed we'll be good to go :)

Don't forget the upgrade script~

**Comment by datonefaridze** [comment]
@thiryr i corrected everything, can you check again?

File: addons/event/report/event_event_templates.xml
Comment (thiryr) [inline addons/event/report/event_event_templates.xml:66]: This fails because this is a report on event.event, so object doesn't have a barcode. 'attendee' does

File: addons/event/__manifest__.py
Comment (thiryr) [inline addons/event/__manifest__.py:40]: This should be in the MOV commit

File: addons/event/models/res_config_settings.py
Comment (thiryr) [inline addons/event/models/res_config_settings.py:27]: This should also be in the MOV commit

File: addons/event/report/event_event_templates.xml
Comment (thiryr) [inline addons/event/report/event_event_templates.xml:161]: Same as before, we still want to show it in the reports if there's no attendee. Let's use the same condition

**Comment by thiryr** [comment]
Also would be nice to add barcode values in the demo data, for demo purposes

File: addons/event/models/event_registration.py
Comment (thiryr) [inline addons/event/models/event_registration.py:11]: Useless change

File: addons/event/data/mail_template_data.xml
Comment (thiryr) [inline addons/event/data/mail_template_data.xml:275]: Wouldn't have empty lines here

**Review by reth-odoo — COMMENTED** [review]
Seems ok, just a few empty line comments and I think we're good to go once it's green

Note that upgrade script will probably be annoying about styling, try to check the 'matt' CI on there to see the exact issues it has.

**Review by reth-odoo — APPROVED** [review]
Seems ok now, just this last :socks: 

Thanks for the changes!

File: addons/event/models/event_registration.py
Comment (datonefaridze) [reply addons/event/models/event_registration.py:23]: of course, this bastard won't last too long

File: addons/event/models/event_registration.py
Comment (datonefaridze) [reply addons/event/models/event_registration.py:32]: hmm, that's a good question

Comment (Pirols) [reply addons/event/models/event_registration.py:32]: Maybe test this (e.g. change it to cause some collisions and observe the behavior), but I don't think it will behave "optimally" on its own. Maybe it'd be better to check if the randomly generated value is already in use inside this default function.

Comment (datonefaridze) [reply addons/event/models/event_registration.py:32]: @Pirols asked ORM team, they said that it will just raise an error, so when people see the error they can re-generate it again

Comment (HydrionBurst) [reply addons/event/models/event_registration.py:32]: the `index=True` might be useless because there is an unique constraint for `unique(name)` which will create a unique btree index in PostgreSQL. Maybe an extra commit to remove it.

File: addons/event/report/event_event_templates.xml
Comment (tde-banana-odoo) [inline addons/event/report/event_event_templates.xml:55]: I would make a use_barcode field on event / registration models, that hides the low-level detail of checking the parameter check. That way you simply do "t-if='attendee.use_barcode' ..." and voila.

File: addons/event/static/src/client_action/event_barcode.js
Comment (tde-banana-odoo) [inline addons/event/static/src/client_action/event_barcode.js:35]: Could be renamed to eventData, to ease understanding its usage in the file :)

File: addons/event/static/src/client_action/event_barcode.js
Comment (tde-banana-odoo) [inline addons/event/static/src/client_action/event_barcode.js:80]: You can probably use the 'action_event_view' action that exists, unless you would have reason not to do so ? that way we have less custom action and we rely on existing xml records ...

File: addons/event/views/event_event_views.xml
Comment (tde-banana-odoo) [inline addons/event/views/event_event_views.xml:3]: :socks: could be put into a 'ir_actions_client_data_xml' data file in data

**Review by tde-banana-odoo — COMMENTED** [review]
Pre-review to speedup ! Mainly some comments on moved code (in order to improve code, probably to do in separate commit(s) ). Other code seems good, will have another look after update :)

File: addons/event/models/event_registration.py
Comment (datonefaridze) [reply addons/event/models/event_registration.py:11]: i assume `` was mistake here

File: addons/event/views/event_event_views.xml
Comment (datonefaridze) [reply addons/event/views/event_event_views.xml:3]: @tde-banana-odoo there doesn't exist any file with such name, and i don't think creating it would be better option than leaving this here

File: addons/event/static/src/client_action/event_barcode.js
Comment (datonefaridze) [reply addons/event/static/src/client_action/event_barcode.js:80]: @tde-banana-odoo i tried it but then it puts me into this wierd view
`this.actionService.doAction("event.action_event_view");`

image

Comment (datonefaridze) [reply addons/event/static/src/client_action/event_barcode.js:80]: i mean there are no menus available at the top

Comment (datonefaridze) [reply addons/event/static/src/client_action/event_barcode.js:80]: refreshing helps but i don't think that's a correct flow
image

Comment (datonefaridze) [reply addons/event/static/src/client_action/event_barcode.js:80]: i guess that's what the comment was saying, so i will leave this one

File: addons/event/static/src/client_action/event_barcode.js
Comment (datonefaridze) [reply addons/event/static/src/client_action/event_barcode.js:35]: @tde-banana-odoo I think changing the name won't work, since this method overrides the parent

**Comment by datonefaridze** [comment]
@tde-banana-odoo can i get a r+ ? until somebody makes changes to master and i have to rebase again :D

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**
